feat: recommendation caculation logic

betterb2etr opened the pull request:
## **PR 타이틀**

- Squash 시 최종 커밋 제목이 되므로 **Conventional Commits** 권장
    
    예) PR 제목 (#PR번호), `feat: support refresh token rotation (#1234)`

##  Summary
- 무엇/왜 (한 줄 요약)
- application : 밖으로 보내는 추천될 강의들 정보 
- domain/model/ids : upstream에서 받아오는 데이터들 
- domain/port : 받아오는 데이터들 필터링
- domain/servie : 추천 강의들 점수 계산 로직 
전반적으로 어떻게 데이터를 받아올지를 결정하고 점수를 계산하는 코드를 작성했습니다. 

##  Related Issue
- Issue Number: #<이슈번호>
- (선택) Closes #<이슈번호>  ← 머지 시 자동 닫힘

##  Changes
- 주요 변경사항 bullet
-upstream (user, course, enrollment)에서 정보를 받아오는 로직들을 구축했습니다. 
 사용자에게 추천할 강의 후보군은 약 100개로 설정하였습니다. 

##  Discussion Topic
- 의논하고싶은 주제
LXP에 등록된 강좌들이 200개가 넘어가면 사용자가 수강중인 강좌 필터링을 위한 로직을 별도로 구현해야 합니다. 
##  Checklist
- [ ] 로컬 테스트 완료
- [ ] lint 통과(로컬)
- [ ] 리뷰어 요청(최소 2명), 라벨/스코프 지정
- [ ] 브랜치 최신화(rebase) 및 충돌 없음

betterb2etr committed:
feat: Entity, Repository

betterb2etr committed:
feat: Entity, Repository

betterb2etr committed:
feat: Entity, Repository

betterb2etr committed:
feat: Entity, Repository

# Conflicts:
#	recommend/build.gradle
#	recommend/src/main/java/com/lxp/recommend/domain/model/MemberRecommendation.java
#	recommend/src/main/java/com/lxp/recommend/domain/model/RecommendedCourse.java
#	recommend/src/main/java/com/lxp/recommend/domain/repository/MemberRecommendationRepository.java
#	recommend/src/main/java/com/lxp/recommend/infrastructure/persistence/JpaMemberRecommendationRepository.java

betterb2etr committed:
feat: Entity, Repository

betterb2etr committed:
feat: Entity, Repository

betterb2etr committed:
feat: Entity, Repository

betterb2etr committed:
feat: Entity, Repository

betterb2etr committed:
feat: Recommendation Controller

betterb2etr committed:
feat: Recommendation Controller

betterb2etr committed:
feat: Recommendation Controller

betterb2etr committed:
feat: Recommendation Controller

betterb2etr committed:
feat: Recommendation Controller

betterb2etr committed:
feat: Recommendation Controller

betterb2etr committed:
feat: Recommendation Controller

betterb2etr committed:
feat: Recommendation Controller

betterb2etr committed:
Merge remote-tracking branch 'origin/seohyeon/feat/recommendation' into seohyeon/feat/recommendation

minyong96 committed:
hotfix: delete setting.gradle (#9)

## **PR 타이틀**

- Squash 시 최종 커밋 제목이 되므로 **Conventional Commits** 권장
    
    예) PR 제목 (#PR번호), `feat: support refresh token rotation (#1234)`

##  Summary
- 무엇/왜 (한 줄 요약)

##  Related Issue
- Issue Number: #<이슈번호>
- (선택) Closes #<이슈번호>  ← 머지 시 자동 닫힘

##  Changes
- 주요 변경사항 bullet

##  Discussion Topic
- 의논하고싶은 주제

##  Checklist
- [ ] 로컬 테스트 완료
- [ ] lint 통과(로컬)
- [ ] 리뷰어 요청(최소 2명), 라벨/스코프 지정
- [ ] 브랜치 최신화(rebase) 및 충돌 없음

betterb2etr committed:
Merge remote-tracking branch 'origin/seohyeon/feat/recommendation' into seohyeon/feat/recommendation

betterb2etr committed:
fix: UUID -> Long

betterb2etr committed:
fix: UUID -> Long

betterb2etr committed:
fix: UUID -> Long

betterb2etr committed:
fix: UUID -> Long

betterb2etr committed:
feat: port/ readers and view

betterb2etr committed:
feat: port/ readers and view

betterb2etr committed:
feat: port/ readers and view

betterb2etr committed:
feat: port/ readers and view

betterb2etr committed:
refact: add explaination

betterb2etr committed:
refact: add explaination

betterb2etr committed:
refact: add explaination

betterb2etr committed:
refact: add explaination

betterb2etr committed:
docs: add readme.md to prepare 1204(absent)

betterb2etr committed:
docs: add readme.md to prepare 1204(absent)

betterb2etr committed:
docs: add readme.md to prepare 1204(absent)

betterb2etr committed:
docs: add readme.md to prepare 1204(absent)

betterb2etr committed:
docs: readme.md for 1204(absent)

betterb2etr committed:
docs: readme.md for 1204(absent)

betterb2etr committed:
docs: readme.md for 1204(absent)

betterb2etr committed:
docs: readme.md for 1204(absent)

betterb2etr committed:
feat: making candidates by careerType

betterb2etr committed:
feat: making candidates by careerType

betterb2etr committed:
feat: making candidates by careerType

betterb2etr committed:
feat: making candidates by careerType

betterb2etr committed:
fix: reflects differences (tags)

betterb2etr committed:
fix: reflects differences (tags)

betterb2etr committed:
fix: reflects differences (tags)

betterb2etr committed:
fix: reflects differences (tags)

betterb2etr committed:
fix: id) long -> String

betterb2etr committed:
fix: id) long -> String

betterb2etr committed:
fix: id) long -> String

betterb2etr committed:
fix: id) long -> String

betterb2etr committed:
Merge branch 'seohyeon/feat/recommendation' of https://github.com/Poten-Up-3rd-Project/LXP-Project-3rd into seohyeon/feat/recommendation

dnya0 committed:
feat(user, auth): define User port (#26)

## **PR 타이틀**

- Squash 시 최종 커밋 제목이 되므로 **Conventional Commits** 권장
    
    예) PR 제목 (#PR번호), `feat: support refresh token rotation (#1234)`

##  Summary
- 무엇/왜 (한 줄 요약)

##  Related Issue
- Issue Number: #<이슈번호>
- (선택) Closes #<이슈번호>  ← 머지 시 자동 닫힘

##  Changes
- User의 port, Auth의 JwtPolicy 및 UserCase 정의
- api 문서 추가

##  Discussion Topic
- 더 추가 또는 수정할 것이 있을지..

##  Checklist
- [x] 로컬 테스트 완료
- [x] lint 통과(로컬)
- [x] 리뷰어 요청(최소 2명), 라벨/스코프 지정
- [x] 브랜치 최신화(rebase) 및 충돌 없음

betterb2etr committed:
docs: preparing for refact

betterb2etr committed:
docs: preparing for refact

betterb2etr committed:
fix : Long -> String, limit candidates 100

betterb2etr committed:
fix : Long -> String, limit candidates 100

betterb2etr committed:
fix: long ->string (UUID)

betterb2etr committed:
fix: long ->string (UUID)

betterb2etr committed:
docs: api docs

betterb2etr committed:
docs: api docs

minyong96 commented on recommend/build.gradle:
```diff
@@ -21,7 +21,12 @@ repositories {
 dependencies {
+    implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
+    compileOnly 'org.projectlombok:lombok'
+    annotationProcessor 'org.projectlombok:lombok'
```
이 부분 root에 의존성 추가 되어있어서 제거해줘도 될 거 같습니다.

minyong96 commented on recommend/build.gradle:
```diff
@@ -21,7 +21,12 @@ repositories {
+    annotationProcessor 'org.projectlombok:lombok'
 
+    // Controller, @RestController, ResponseEntity 등을 위해 필요
+    implementation 'org.springframework.boot:spring-boot-starter-web'
```
이부분도요!

betterb2etr replied:
org.springframework.boot:spring-boot-starter-web' 의 경우,  루트에 없어서 그대로 가는 것으로 하겠습니다.

betterb2etr replied:
지우니까 어노테이션을 사용을 못하더라구요

minyong96 commented:
이 부분 내일 전체적인 흐름 설명해주실수있을까여?

minyong96 commented on recommend/readme.md:
```diff
@@ -0,0 +1,282 @@
+│   ├─ port (Interface)
+│   │   ├─ MemberProfileReader
+│   │   ├─ CourseMetaReader
+│   │   └─ LearningStatusReader
```
이 부분은 외부 BC에서 읽어 오는 부분이죠? 
지금 이 부분은 외부 BC 데이터를 읽어오는 로직인데, Port 인터페이스가 도메인에 들어가 있어서 구조가 어긋났는데,
도메인의 책임은 순수 비즈니스 규칙만 유지하는 것이기 때문이기에 도메인은 절대로 외부 시스템(다른 BC, 외부 API, DB, 메시지 브로커 등)에 의존하면 안 됩니다

Port는 애플리케이션 레이어에서 정의해야 하고,
외부 BC와의 통신 구현체는 infra/adapter에서 Port를 구현해야 합니다.

즉,
- application/required 에 외부에서 가져올 Port 인터페이스를 정의하고
- infra/adapter 에서 해당 Port를 구현한 Adapter를 만들고, 외부에서 제공하는 port를 가져다 쓰셔야돼요

이렇게 해야 도메인이 외부에 독립적이고,
헥사고날 아키텍처 의존성 방향을 유지할 수 있습니당

betterb2etr replied:
application/required 에 외부에서 가져올 Port 인터페이스 정의 완료 했습니다. , 외부 BC 와의 통신 구현체는 머지되고 난다음에 진행하겠습니다.  Adapter 구현체는 Member/Course BC와의 연동 시점에 추가 예정입니다. 현재는 인터페이스만 정의하여 구조를 확립했습니다

minyong96 committed:
fix: common-lib version up

betterb2etr committed:
fix: duplicated setting delete

betterb2etr committed:
fix: duplicated setting delete

betterb2etr committed:
refact: domain/ port-> app/required

betterb2etr committed:
refact: domain/ port-> app/required

betterb2etr committed:
refact: app/required -> app/port/required

betterb2etr committed:
refact: app/required -> app/port/required

betterb2etr committed:
fix: add web dependencies

betterb2etr committed:
fix: add web dependencies

betterb2etr committed:
fix : add dependencies

betterb2etr committed:
fix : add dependencies

minyong96 committed:
chore: fix module (#38)

minyong96 commented on recommend/src/main/java/com/lxp/recommend/application/RecommendationApplicationService.java:
```diff
@@ -0,0 +1,121 @@
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class RecommendationApplicationService {
```
port 나누기, 그리고 하나의 책임만 하는 service 나누기

minyong96 commented on recommend/src/main/java/com/lxp/recommend/application/RecommendationApplicationService.java:
```diff
@@ -0,0 +1,121 @@
+                course.getCourseId().getValue(),
+                course.getScore(),
+                course.getRank()
+        );
```
usecase interface 로 호출 (adapter -> service
)

youabledev committed:
docs(enrollment): enrollment swagger documentation (#28)

##  Changes
- 수강 이력 조회
- 수강 신청
- 수강 취소

youabledev committed:
hotfix(application): modulith error fix (#36)

##  Changes
- springdoc-openapi 2.5.0 버전이 spring boot 4.0.0 버전과 맞지 않아
NoSuchMethodError가 발생. springdoc-openapi 3.0.0으로 변경
- MainApplication가 패키지 com.lxp.application 에 위치해 있었습니다. 다른 모듈은 com.lxp
하위에 정의되어 있어 컴포넌트 스캔이 제대로 이뤄지지 않고 있었습니다. MainApplicaton 위치가 com.lxp로
변경되었습니다.
- RetryConfig를 구현하는 빈이 없어서 컴포넌트 스캔시 에러가 발생했습니다. 임시로 RetryConfig를 구현해 추가해
두었습니다.
- ApplicationModuleTests 테스트시 com.lxp 를 기준으로 확인할 수 있게 코드를 수정했습니다.
- ApplicationModuleTests 결과 다른 모듈들이 common을 의존하고 있습니다. common이 public하게
다른 모듈에서 사용할 수 있음을 명시해야 해서 package-info.java가 추가되었습니다.

betterb2etr committed:
fix: careerType -> LearnerLevel

betterb2etr committed:
fix: careerType -> LearnerLevel

youabledev commented on recommend/readme.md:
```diff
@@ -0,0 +1,308 @@
+**목표:** LXP(Learning Experience Platform)의 개인화 추천 기능 구현  
+**담당:** Recommendation Bounded Context  
+**기술 스택:**
+- Java 17, Spring Boot 3.x, MySQL 8.0, JPA, Gradle
```
백엔드 내부적으로는 spring boot 4.0.0, mysql 9.0(latest) 사용되고 있습니다!

youabledev commented on recommend/readme.md:
```diff
@@ -0,0 +1,322 @@
+
+### 3. 설계 결정사항
+
+#### (1) ID 타입: UUID → **Long으로 변경**
+- 팀 회의 결정: 성능과 관리 편의성을 위해 `BIGINT AUTO_INCREMENT` 사용
```
user, course는 UUID 사용되고 있는데 협의된 내용이 있었었나요?

betterb2etr replied:
upstream인 user, course 가 식별자 UUID 사용하고 있었고, downstream인 추천도 그에 맞춰가는게 맞다고 생각했습니다. 보안/식별상 Long 보다는 uuid 가 맞기도 하구요.  단, 추천은 식별자가 외부에 노출될 일이 거의 없어서 그대로 id 값 Long으로 가져갑니다.

betterb2etr replied:
tag도 외부에 노출될 일이 없어서 Long으로 그대로 사용하여도 문제가 없을거 같더라구요

youabledev commented on recommend/src/main/java/com/lxp/recommend/application/dto/RecommendedCourseDto.java:
```diff
@@ -0,0 +1,7 @@
+package com.lxp.recommend.application.dto;
+
+public record RecommendedCourseDto(
+        String courseId,
+        double score,
+        int rank
```
controller에 정의된 api에서 
response응답으로 RecommendedCourseDto가 사용되는 것으로 보입니다
프론트 측에서 이 세가지 값만 필요한 건지, course의 모든 정보가 함께 필요한 건지 확인해 봐야 할 것 같습니다

betterb2etr replied:
아직 프론트상에는 score, rank까지는 노출이 안될거 같기는 한데 임의로 담아놓기는 했습니다.

betterb2etr replied:
해당 내용은 리팩토링할 내용으로 분류하겠습니다.

youabledev commented on recommend/src/main/java/com/lxp/recommend/domain/dto/LearnerProfileView.java:
```diff
@@ -0,0 +1,9 @@
+
+public record LearnerProfileView(
+        String memberId,
+        Set<String> selectedTags,
```
태그의 이름 리스트가 필요한 건가요? 
멤버에서는 태그의 id를 리스트로 가지고 있지 않을까 싶어 확인차 문의 드렸습니다!

betterb2etr replied:
아 한번 확인해보겠습니다. 일단 로직은 태그 이름을 상대로 비교하고 있습니다.

youabledev commented on recommend/src/main/java/com/lxp/recommend/domain/model/ids/CourseId.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
record를 사용하면 불변 객체로 정의되어 equals나 getter를 따로 작성하지 않아도 되서 편리하더라구요!

youabledev commented on recommend/src/main/java/com/lxp/recommend/domain/model/MemberRecommendation.java:
```diff
@@ -0,0 +1,74 @@
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity
```
내부에서 의논한 바로는 
model 레이어 영역은 다른 프레임워크나 라이브러리 의존 없이 순수 java 로 구현되어야 하는 것으로 알고있습니다! 
infrastructure에서 entity를 구현하면 어떨까 싶습니다!

betterb2etr replied:
민영님도 언급해주셔서 말씀하신 사항 반영하여 리팩토링중입니다. 리뷰 감사합니답

dnya0 committed:
hotfix(user): resolve parsing error in openapi-user.yml (#41)

## **PR 타이틀**

- Squash 시 최종 커밋 제목이 되므로 **Conventional Commits** 권장
    
    예) PR 제목 (#PR번호), `feat: support refresh token rotation (#1234)`

##  Summary
- swagger 문서 파싱 오류 해결 

##  Related Issue
- Issue Number: #<이슈번호>
- (선택) Closes #<이슈번호>  ← 머지 시 자동 닫힘

##  Changes
- 주요 변경사항 bullet

##  Discussion Topic
- 의논하고싶은 주제

##  Checklist
- [ ] 로컬 테스트 완료
- [ ] lint 통과(로컬)
- [ ] 리뷰어 요청(최소 2명), 라벨/스코프 지정
- [ ] 브랜치 최신화(rebase) 및 충돌 없음

betterb2etr committed:
feat: RecommendException

betterb2etr committed:
feat: RecommendException

betterb2etr committed:
feat: exeption docs

betterb2etr committed:
feat: exeption docs

betterb2etr committed:
config: presentation -> interfaces

betterb2etr committed:
config: presentation -> interfaces

betterb2etr committed:
config: create jpa/ dir

betterb2etr committed:
config: create jpa/ dir

betterb2etr committed:
fix: separate jpa

betterb2etr committed:
fix: separate jpa

betterb2etr committed:
config: separate jpa

betterb2etr committed:
config: separate jpa

betterb2etr committed:
refact: separate VO

betterb2etr committed:
refact: separate VO

betterb2etr committed:
fix: fix naming recommendation -> recommend

betterb2etr committed:
fix: fix naming recommendation -> recommend

betterb2etr committed:
refact: recommendApplicationService

betterb2etr committed:
refact: recommendApplicationService

minyong96 committed:
feat: createCourse Event (#42)

## **PR 타이틀**

- Squash 시 최종 커밋 제목이 되므로 **Conventional Commits** 권장
    
    예) PR 제목 (#PR번호), `feat: support refresh token rotation (#1234)`

##  Summary
- 무엇/왜 (한 줄 요약)

##  Related Issue
- Issue Number: #<이슈번호>
- (선택) Closes #<이슈번호>  ← 머지 시 자동 닫힘

##  Changes
- 주요 변경사항 bullet

##  Discussion Topic
- 의논하고싶은 주제

##  Checklist
- [ ] 로컬 테스트 완료
- [ ] lint 통과(로컬)
- [ ] 리뷰어 요청(최소 2명), 라벨/스코프 지정
- [ ] 브랜치 최신화(rebase) 및 충돌 없음

dnya0 committed:
feat(user, auth): implement JPA Persistence Adapter for User And Auth Aggregate (#39)

betterb2etr committed:
feat: domain/model test codes

betterb2etr committed:
feat: domain/model test codes

betterb2etr committed:
docs: readme.md preparing for 1211

betterb2etr committed:
docs: readme.md preparing for 1211

betterb2etr committed:
docs: readme.md preparing for 1211

Reorganize package structure and clarify design principles.

betterb2etr committed:
docs: readme.md preparing for 1211

Reorganize package structure and clarify design principles.

minyong96 committed:
Hotfux/api (#45)

## **PR 타이틀**

- Squash 시 최종 커밋 제목이 되므로 **Conventional Commits** 권장
    
    예) PR 제목 (#PR번호), `feat: support refresh token rotation (#1234)`

##  Summary
- 무엇/왜 (한 줄 요약)

##  Related Issue
- Issue Number: #<이슈번호>
- (선택) Closes #<이슈번호>  ← 머지 시 자동 닫힘

##  Changes
- 주요 변경사항 bullet

##  Discussion Topic
- 의논하고싶은 주제

##  Checklist
- [ ] 로컬 테스트 완료
- [ ] lint 통과(로컬)
- [ ] 리뷰어 요청(최소 2명), 라벨/스코프 지정
- [ ] 브랜치 최신화(rebase) 및 충돌 없음

---------

Co-authored-by: youable <youable.framios@gmail.com>
Co-authored-by: dnya0 <annay01794@gmail.com>

dnya0 committed:
feat(user): update user info and status ports to return Optional types (#46)

## **PR 타이틀**

- Squash 시 최종 커밋 제목이 되므로 **Conventional Commits** 권장
    
    예) PR 제목 (#PR번호), `feat: support refresh token rotation (#1234)`

##  Summary
- 무엇/왜 (한 줄 요약)

##  Related Issue
- Issue Number: #<이슈번호>
- (선택) Closes #<이슈번호>  ← 머지 시 자동 닫힘

##  Changes
- User의 ExternalPort들 구현 및 return 값 Optional로 변경하였습니다.

##  Discussion Topic
- 의논하고싶은 주제

##  Checklist
- [ ] 로컬 테스트 완료
- [ ] lint 통과(로컬)
- [ ] 리뷰어 요청(최소 2명), 라벨/스코프 지정
- [ ] 브랜치 최신화(rebase) 및 충돌 없음

minyong96 committed:
feat(content): apply test and invalid user (#47)

minyong96 commented on recommend/src/main/java/com/lxp/recommend/application/service/RecommendApplicationService.java:
```diff
@@ -0,0 +1,202 @@
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class RecommendApplicationService {
```
Service에 책임이 너무 많습니당
책임 분리해주세용

betterb2etr replied:
아래 comment 남겨주신대로, common의 enum을 활용하는 방향으로 전환하고, 책임분리하겠습니다. 1212 작업 진행중입니다.

minyong96 commented on recommend/src/main/java/com/lxp/recommend/domain/model/MemberRecommendation.java:
```diff
@@ -0,0 +1,155 @@
+ * 1. 추천 아이템의 생명주기 관리
+ * 2. 불변식 보장
+ */
+public class MemberRecommendation extends AggregateRoot {
```
AggregateRoot의 Id타입추가해주세요!

minyong96 commented on recommend/src/main/java/com/lxp/recommend/domain/model/MemberRecommendation.java:
```diff
@@ -0,0 +1,155 @@
+    protected MemberRecommendation() {} // JPA 및 재구성용
+
+    public MemberRecommendation(MemberId memberId) {
+        if (memberId == null) {
+            throw new IllegalArgumentException("memberId는 null일 수 없습니다.");
+        }
```
앞쪽 즉 application layer에서 이미 검증을 하니 개발자의 오류를 위함이라면 Objects.requireNonNull 로 바꿔주시면 더 좋을듯 합니다.

minyong96 commented on recommend/src/main/java/com/lxp/recommend/domain/model/MemberRecommendation.java:
```diff
@@ -0,0 +1,155 @@
+        }
+
+        // 불변식 1: 최대 10개
+        if (items.size() > MAX_RECOMMENDATION_SIZE) {
+            throw new RecommendLimitExceededException(
+                    String.format("추천 아이템은 최대 %d개까지 가능합니다. (현재: %d개)",
+                            MAX_RECOMMENDATION_SIZE, items.size())
+            );
+        }
+
+        // 불변식 2: 중복 강좌 없음
+        Set<CourseId> courseIds = new HashSet<>();
+        for (RecommendedCourse item : items) {
+            if (!courseIds.add(item.getCourseId())) {
+                throw new DuplicateCourseException(
+                        "중복된 강좌가 포함되어 있습니다: " + item.getCourseId().getValue()
+                );
+            }
+        }
+
+        // 불변식 3: 순위 연속성 검증 (1, 2, 3, ...)
+        for (int i = 0; i < items.size(); i++) {
+            int expectedRank = i + 1;
+            int actualRank = items.get(i).getRank();
+            if (actualRank != expectedRank) {
+                throw new IllegalArgumentException(
+                        String.format("순위가 연속적이지 않습니다. 기대: %d, 실제: %d", expectedRank, actualRank)
```
불변식 함수 나눠주시면 더 좋을듯 합니다.
valid 쪼개기

minyong96 commented on recommend/src/main/java/com/lxp/recommend/domain/model/MemberRecommendation.java:
```diff
@@ -0,0 +1,155 @@
+     * 점수 계산 후 사용되는 임시 VO
+     * (도메인 서비스 → Aggregate 전달 시 사용)
+     */
+    public record ScoredCourse(
+            CourseId courseId,
+            double score
+    ) {
```
이너 클래스로 두신 이유가 있을까여?

minyong96 commented on recommend/src/main/java/com/lxp/recommend/application/service/RecommendApplicationService.java:
```diff
@@ -0,0 +1,202 @@
+            return;
+        }
+
+        Set<DifficultyLevel> targetDifficulties = determineTargetDifficulties(profile.learnerLevel());
```
서현님 이 부분 common에 정의해뒀는데 Course쪽에 CourseDifficulty가 아닌 Level로 관리될거같아
매퍼함수 불필요할거같습니당

minyong96 commented:
서현님. 지금 develop 브랜치가 많이 수정되어서, rebase 한번 진행 해주시고 작업 계속 해주실래여?

betterb2etr committed:
config: adapter dir location chage

betterb2etr committed:
config: adapter dir location chage

betterb2etr committed:
docs: 1212

betterb2etr committed:
docs: 1212

betterb2etr committed:
feat: port DTO

betterb2etr committed:
feat: port DTO

betterb2etr committed:
Merge branch 'seohyeon/feat/recommendation' of https://github.com/Poten-Up-3rd-Project/LXP-Project-3rd into seohyeon/feat/recommendation

betterb2etr committed:
fix: add import

betterb2etr committed:
feat: adapter, mapper

betterb2etr committed:
fix: view recommend result

betterb2etr committed:
feat: application tests

betterb2etr committed:
fix: test

betterb2etr committed:
fix: apply level of common

betterb2etr committed:
fix: separate
